receiver/k8sclusterreceiver/metadatastore.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sclusterreceiver
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/cache"
+)
+
+// metadataStore keeps track of required caches exposed by informers.
+// This store is used while collecting metadata about Pods to be able
+// to correlate other Kubernetes objects with a Pod.
+type metadataStore struct {
+	stores map[string]cache.Store
+}
+
+// setupStore tracks metadata of services, jobs and replicasets.
+func (ms *metadataStore) setupStore(o runtime.Object, store cache.Store) {
+	switch o.(type) {
+	case *corev1.Service:
+		ms.stores["Service"] = store
+	case *batchv1.Job:
+		ms.stores["Job"] = store
+	case *appsv1.ReplicaSet:
+		ms.stores["ReplicaSet"] = store
```

Comment:
Store names like "Service" appear to be scattered across the codebase. It will be very easy for a typo to result in a bug. I think it will be better to define a type alias for store name and define consts for 3 store names.

Reply:
Added constants for `Kind`s in `collector.go`